tests/test_utils.py

```diff
@@ -0,0 +1,181 @@
+from unittest.mock import patch, MagicMock
+from app.etl.generate_perf_report import GeneratePerfReport
+from app.etl.generate_recon_report import GenerateReconReport
+from app.etl.publish_data import PublishData
+from helpers.utils import (extract_mm_dd_yyyy, extract_yyyy_mm_dd, extract_yyyymmdd, extract_dd_mm_yyyy, extract_report_date,)
+from app.etl.pre_process_data import PreprocessData
+import polars as pl
+
```

Comment:
Would be good to see some tests that check that the appropriate exception is raised during an error condition.